File: tp6/python/backend-python/src/controllers/matrices_controller.py
from flask import Request

from src.models.matriz_model import multiplicar_matrices, sumar_matrices


def sumar_controller(request: Request):
    body = request.get_json(silent=True) or {}

    matriz1 = body.get("matriz1")
    matriz2 = body.get("matriz2")

    if matriz1 is None or matriz2 is None:
        return {
            "exito": False,
            "error": "Ambas matrices son requeridas",
        }, 400

    resultado = sumar_matrices(matriz1, matriz2)
    status = 200 if resultado.get("exito") else 400
    return resultado, status


def multiplicar_controller(request: Request):
    body = request.get_json(silent=True) or {}

    matriz1 = body.get("matriz1")
    matriz2 = body.get("matriz2")

    if matriz1 is None or matriz2 is None:
        return {
            "exito": False,
            "error": "Ambas matrices son requeridas",
        }, 400

    resultado = multiplicar_matrices(matriz1, matriz2)
    status = 200 if resultado.get("exito") else 400
    return resultado, status

File: tp6/js/matrices-backend/src/controllers/matrices.controller.js
import {
  sumarMatrices,
  multiplicarMatrices,
} from "../models/matriz.model.js";

export const sumar = (req, res) => {
  try {
    const { matriz1, matriz2 } = req.body;

    if (!matriz1 || !matriz2) {
      return res.status(400).json({
        exito: false,
        error: "Ambas matrices son requeridas",
      });
    }

    const resultado = sumarMatrices(matriz1, matriz2);
    res.json(resultado);
  } catch (error) {
    res.status(500).json({
      exito: false,
      error: "Error en el servidor: " + error.message,
    });
  }
};

export const multiplicar = (req, res) => {
  try {
    const { matriz1, matriz2 } = req.body;

    if (!matriz1 || !matriz2) {
      return res.status(400).json({
        exito: false,
        error: "Ambas matrices son requeridas",
      });
    }

    const resultado = multiplicarMatrices(matriz1, matriz2);
    res.json(resultado);
  } catch (error) {
    res.status(500).json({
      exito: false,
      error: "Error en el servidor: " + error.message,
    });
  }
};
